bool move
int32 existence_count
char name
